### PR TITLE
Fix an issue where alwaysOnTop window disappears when restore…

### LINKED ISSUE #9170: alwaysOnTop window disappears when restored from the application icon
* Electron version: 1.6.2
* Operating system: Mac OS Serra 

### Expected behavior

Restoring browser window with always on top should restore 

### Actual behavior

with always on top set and minimze window into application icon window restoring a minimized window will show and then disappear and lost forever. 

### How to reproduce

  $ git clone https://github.com/jschimmoeller/app-icon-consolidate.git
  $ npm install
  $ npm start 

make sure Minize windows into application icon is set 
image

steps:  
1.  minimize app
2.  click on electron icon in dock to restore 
3.  window will restore and then disappear

### PR DESCRIPTION
**Pull request by shubham2892**
Fixes #9170 
Reset the custom level when windowWillMiniaturize fires and then restores it when windowDidDeminiaturize fires as suggested in issue.

Before: 
Before Gif

After:
After Gif

### CODE REVIEW

File: atom/browser/native_window_mac.mm [hunk 1]
```diff
@@ -204,11 +205,20 @@ - (void)windowDidMove:(NSNotification*)notification {
   shell_->NotifyWindowMoved();
 }
 
+- (void)windowWillMiniaturize:(NSNotification*)notification {
+  NSWindow* window = shell_->GetNativeWindow();
+  // store the current status window level to be restored in windowDidDeminiaturize
+  level_ = NSStatusWindowLevel;
```
Comment (shubham2892) [inline atom/browser/native_window_mac.mm]: Yes, It should. My bad. Fixed it.

File: atom/browser/native_window_mac.mm [hunk 2]
```diff
@@ -75,6 +75,7 @@ @interface AtomNSWindowDelegate : NSObject<NSWindowDelegate> {
  @private
   atom::NativeWindowMac* shell_;
   bool is_zooming_;
+  int level_;
```
Comment (kevinsawicki) [inline atom/browser/native_window_mac.mm]: I think this should be initialized in `initWithShell` like the other variables.

File: spec/api-browser-window-spec.js [hunk 3]
```diff
@@ -620,6 +620,22 @@ describe('BrowserWindow module', function () {
     })
   })
 
+  describe.only('BrowserWindow.alwaysOnTop() resets level on minimize', function () {
```
Comment (kevinsawicki) [inline spec/api-browser-window-spec.js]: :fire: `.only`

**Comment by kevinsawicki** [comment]
Thanks for fixing this @shubham2892 👍 🚢

**Pull request merged**